gcloud/bigtable/cluster.py

```diff
@@ -15,9 +15,51 @@
 """User friendly container for Google Cloud Bigtable Cluster."""
 
 
+import re
+
 from gcloud.bigtable.table import Table
 
 
+_CLUSTER_NAME_RE = re.compile(r'^projects/(?P<project>[^/]+)/'
+                              r'zones/(?P<zone>[^/]+)/clusters/'
+                              r'(?P<cluster_id>[a-z][-a-z0-9]*)$')
+
+
+def _require_pb_property(message_pb, property_name, value):
+    """Check that a property agrees with the value on the message.
+
+    :type message_pb: :class:`google.protobuf.message.Message`
+    :param message_pb: The message to check for ``property_name``.
+
+    :type property_name: str
+    :param property_name: The property value to check against.
+
+    :type value: object or :data:`NoneType <types.NoneType>`
+    :param value: The value to check against the cluster. If :data:`None`,
+                  will not be checked.
+
+    :rtype: object
+    :returns: The value of ``property_name`` set on ``message_pb``.
+    :raises: :class:`ValueError <exceptions.ValueError>` if the result returned
+             from the ``message_pb`` does not contain the ``property_name``
+             value or if the value returned disagrees with the ``value``
+             passed with the request (if that value is not null).
+    """
+    # Make sure `property_name` is set on the response.
+    # NOTE: HasField() doesn't work in protobuf>=3.0.0a3
```

Comment:
The [generated docs](https://developers.google.com/protocol-buffers/docs/reference/python/google.protobuf.message.Message-class#HasField) say that `HasField` will raise `ValueError` if the message doesn't have such a field defined, and otherwise returns True IFF the field is non-empty.  OTOH, the [more modern looking version](https://developers.google.com/protocol-buffers/docs/reference/python-generated#message) say:

> Unlike in proto2, you cannot call HasField() for a singular non-message field in proto3, and the library will throw an exception if you try to do this.